packages/react-button/src/components/Button/useButton.ts

```diff
@@ -1,13 +1,18 @@
-import { tokensToStyleObject } from '@fluentui/react-theme-provider';
+import { ComposePreparedOptions } from '@fluentui/react-compose';
+import { getStyleFromPropsAndOptions } from '@fluentui/react-theme-provider';
 import { ButtonProps, ButtonState } from './Button.types';
 
 /**
  * The useButton hook processes the Button component props and returns state.
  * @param props - Button props to derive state from.
  */
-export const useButton = (props: ButtonProps): ButtonState => {
+export const useButton = (
+  props: ButtonProps,
+  ref: React.Ref<HTMLElement>,
+  options: ComposePreparedOptions,
+): ButtonState => {
   return {
     ...props,
-    style: { ...props.style, ...tokensToStyleObject(props.tokens, '--button') },
+    style: getStyleFromPropsAndOptions(props, options),
```

Comment:
shouldn't we do this in `resolveSlotProps`?

Reply:
We probably should, I wanted to take a stab and solve this in the `Button` and get some feedback before applying it at the compose layer.

Reply:
probably better to keep them separated for now